arch/arc/Kconfig

```diff
@@ -41,6 +41,7 @@ menu "ARCv2 Family Options"
 config	CPU_ARCV2
 	bool
 	select		ARCH_HAS_STACK_PROTECTION
+	select ARCH_HAS_USERSPACE if ARC_CORE_MPU
```

Comment:
Hmm. The problem with exporting CONFIG_ARCH_HAS_USERSPACE is that there is a bunch of userspace test code that will be enabled, but this patch doesn't completely implement userspace for ARC, just the memory domain part. Is the privileged/non-privileged implementation in progress? Might be better incorporate into this patch series, and then only enable ARCH_HAS_USERSPACE in a commit at the end so that the tree remains bisectable.

if the rest of the userspace support is not forthcoming for a while, I would remove this and add this later once everything is in place. otherwise our QA testing will suddenly get a ton of errors.

Reply:
The full user space is still in progress, and will be ready in the end of 2018.01. As the mem_domain_api requires CONFIG_USERSPACE, and CONFIG_USERSPACE requires ARCH_HAS_USERPSACE. So I add it here like ARM did.  To avoid the errors of QA, maybe it's better to remove it now and add it back when everything is ready